lib/Sema/CSApply.cpp

```diff
@@ -4198,16 +4198,16 @@ namespace {
       if (!method->isObjC()) {
           // If the method declaration lies in a protocol and we're providing
           // a default implementation of the method through a protocol extension
-          // then insert the fixit on protocol, rather than on the method in the
+          // then insert the fix-it on protocol, rather than on the method in the
           // protocol declaration (not allowed).
-          auto protocolDecl = foundDecl->getDeclContext()->getSelfProtocolDecl();
+          auto protocolDecl = dyn_cast<ProtocolDecl>(foundDecl->getDeclContext()->getAsDecl());
```

Comment:
Further nitpick: we've got it set up so you can cast from DeclContext directly to Decl subclasses, so the `getAsDecl()` isn't needed.

Reply:
Oh! Fixed now.